contracts/ReaderFromConfig.sol

```diff
@@ -15,76 +18,47 @@ contract ReaderFromConfig {
     )
         internal
         view
-        returns ( uint256 rv )
+        returns (uint256)
     {
-        uint256 fmp = FREE_MEM_PTR;
-        uint256 blocks = (bytes(strConfigVariableName).length + 31) / 32 + 1;
-        bool success;
-        // solhint-disable-next-line no-inline-assembly
-        assembly {
-            let ptr := mload(fmp)
-            for { let i := 0 } lt( i, blocks ) { i := add(1, i) } {
-                let where := add(ptr, mul(32, i))
-                let what := mload(add(strConfigVariableName, mul(32, i)))
-                mstore(where, what)
-            }
-            success := staticcall(not(0), FN_NUM_GET_CONFIG_VARIABLE_UINT256, ptr, mul( blocks, 32 ), ptr, 32)
-            rv := mload(ptr)
-        }
+        (bool success, bytes memory answer) = FN_NUM_GET_CONFIG_VARIABLE_UINT256.staticcall(
+            bytes(strConfigVariableName)
+        );
         require(success, "Get config uint256 failed");
+        return uint256(bytes32(_bytesToBytes32(answer)));
     }
 
     function _getConfigVariableString(
         string memory strConfigVariableName
     )
         internal
         view
-        returns ( string memory rv )
+        returns (string memory)
     {
-        uint256 fmp = FREE_MEM_PTR;
-        uint256 blocks = (bytes(strConfigVariableName).length + 31) / 32 + 1;
-        bool success;
-        // solhint-disable-next-line no-inline-assembly
-        assembly {
-            let ptr := mload(fmp)
-            for { let i := 0 } lt( i, blocks ) { i := add(1, i) } {
-                let where := add(ptr, mul(32, i))
-                let what := mload(add(strConfigVariableName, mul(32, i)))
-                mstore(where, what)
-            }
-            success := staticcall(
-                not(0),
-                FN_NUM_GET_CONFIG_VARIABLE_STRING,
-                ptr,
-                mul( blocks, 32 ),
-                rv,
-                mul( 1024, 1024 )
-            )
-        }
+        (bool success, bytes memory answer) = FN_NUM_GET_CONFIG_VARIABLE_STRING.staticcall(
+            bytes(strConfigVariableName)
+        );
         require(success, "Get config string failed");
+        return string(answer);
     }
 
     function _getConfigVariableAddress(
         string memory strConfigVariableName
     )
         internal
         view
-        returns ( address rv )
+        returns (address)
     {
-        uint256 fmp = FREE_MEM_PTR;
-        uint256 blocks = (bytes(strConfigVariableName).length + 31) / 32 + 1;
-        bool success;
+        (bool success, bytes memory answer) = FN_NUM_GET_CONFIG_VARIABLE_ADDRESS.staticcall(
+            bytes(strConfigVariableName)
+        );
+        require(success, "Get config address failed");
+        return address(bytes20(_bytesToBytes32(answer)));
+    }
+
+    function _bytesToBytes32(bytes memory data) private pure returns (bytes32 result) {
         // solhint-disable-next-line no-inline-assembly
         assembly {
```

Comment:
Assembly here can be avoided because starting solidity 0.8.5 an ability to convert `bytes` to fixed bytes types was added.

I prefer to rewrite the function in this way:

```solidity
function _bytesToBytes32(bytes memory data) private pure returns (bytes32 result) {
        require(data.length == 32, "Wrong data length");
        return bytes32(data);
    }
```

Reply:
Wow true, overthink on this